### PR TITLE
Add DB append tests

### PR DESCRIPTION
**Pull request by beshleman**
This PR adds support for testing the append of the UEFI DB variable. This was the cause of the KB update bug in Windows.

The test has only been tested on Alpine and it requires `efi-updatevar`. It will fail without `efi-updatevar`.

### CODE REVIEW

File: tests/uefistored/test_auth_var.py
Comment (stormi) [inline tests/uefistored/test_auth_var.py:19]:
```suggestion
# - A Linux UEFI VM to import
```

File: tests/uefistored/test_auth_var.py
Comment (stormi) [inline tests/uefistored/test_auth_var.py:69]: The test this snippet belongs to (github won't let me comment the lines higher) should probably depend on the linux_uefi_vm fixture too, and the `if vm.is_windows` check be removed from it.

File: tests/uefistored/test_auth_var.py
Comment (stormi) [inline tests/uefistored/test_auth_var.py:109]: I don't find this much readable for ssh commands, but I won't argue :)

File: tests/uefistored/test_auth_var.py
Comment (benjamreis) [reply tests/uefistored/test_auth_var.py:73]: This rule is not enforced by `pydocstyle`?
Should it?

**Review by benjamreis — COMMENTED** [review]
Nothing to add to Sam review. Will approve once conversation are resolved. :)

File: tests/uefistored/test_auth_var.py
Comment (Wescoeur) [reply tests/uefistored/test_auth_var.py:109]: Why not this instead:
```python
vm.ssh([
    "efi-updatevar",
    "-k",
    vm_kek_key,
    "-c",
    vm_db_cert,
    "db"
])
```

A single level of indentation is quite sufficient. :)

Comment (Wescoeur) [reply tests/uefistored/test_auth_var.py:109]: Or one line: `vm.ssh(["efi-updatevar", "-k", vm_kek_key, "-c", vm_db_cert, "db"])`.
The max char limit is not reached.

File: tests/uefistored/test_auth_var.py
Comment (Wescoeur) [inline tests/uefistored/test_auth_var.py:6]: Maybe `EFI_AUTH` to use a unified syntax?

File: tests/uefistored/test_auth_var.py
Comment (beshleman) [reply tests/uefistored/test_auth_var.py:73]: Why change [PEP257](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings)?

For one-line docstrings, it states:

> The docstring is a phrase ending in a period.

Which is why...

```python
def foo():
    """ Do nothing. """
```

Results in:

```bash
$ pydocstyle /tmp/foo.py

/tmp/foo.py:4 in public function `foo`:
        D210: No whitespaces allowed surrounding docstring text
```

File: tests/uefistored/test_auth_var.py
Comment (beshleman) [reply tests/uefistored/test_auth_var.py:6]: Unfortunately `EFIAuth` is a class and so should probably stay CamelCase, as opposed to the `UPPER_SNAKECASE` variables, which are just "constant" primitives.

I would like to eventually replace the GUIDs with an object that has `as_str()` and `as_bytes()` methods, which will clean up the verbose exports, but that's probably for another PR...

Comment (Wescoeur) [reply tests/uefistored/test_auth_var.py:6]: Oh! I misread the code. :sweat_smile:

File: tests/uefistored/test_auth_var.py
Comment (benjamreis) [reply tests/uefistored/test_auth_var.py:73]: Please rebase your branch on master, this will add new checks regarding code & doc style.

Comment (stormi) [reply tests/uefistored/test_auth_var.py:73]: The current convention in our repositories is with whitespaces (I can find exceptions, such as VM.set_efi_var and VM.file_exists). I don't think we realized PEP257 advised otherwise.

If we want to change that, we'll have to do it globally (in a separate PR), and enforce it in our pydocstyle settings. Regardless of my own taste, it's probably better indeed.

File: tests/uefistored/test_auth_var.py
Comment (beshleman) [reply tests/uefistored/test_auth_var.py:73]: I had to add this a little bit lower, because the VM needs to be booted before being able to detect this.

**Review by stormi — APPROVED** [review]

**Review by benjamreis — APPROVED** [review]
Please rebase on master to add the codestyle checks.

**Comment by stormi** [comment]
Could you rebase on master so that the style checks can run on the PR?

**Comment by beshleman** [comment]
> Please rebase on master to add the codestyle checks.

Done.

Not sure if I should merge or not, so leaving it as is!

**Comment by benjamreis** [comment]
IDK why the doc & code style checks are not run on this PR...
Is there a `main.yaml` & a `format.yaml` in your sources after the rebase? In `.github/workflows`

**Comment by benjamreis** [comment]
Does the commands `pydocstyle` and `pycodestyle` return with no error at the root of the repo on your branch?

**Comment by benjamreis** [comment]
From https://github.com/xcp-ng/xcp-ng-tests/pull/64 the code style is okay.
So this is good to be merged for me.
It seems a simple rebase doesn't trigger the GH actions.

I'll let your merge the Pr when you want :)

**Comment by beshleman** [comment]
> From #64 the code style is okay. So this is good to be merged for me. It seems a simple rebase doesn't trigger the GH actions.
> 
> I'll let your merge the Pr when you want :)

looks good from root:

```
[bobbye@piano xcp-ng-tests]$ pydocstyle
[bobbye@piano xcp-ng-tests]$ pycodestyle
```

Will merge now :)

**Pull request merged**
